exec/kubernetes_deploy.go

```diff
@@ -2,35 +2,90 @@ package exec
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
-	"github.com/portainer/agent"
 	"os"
 	"os/exec"
 	"path"
 	"runtime"
 	"strings"
+
+	"github.com/portainer/agent"
 )
 
 // KubernetesDeployer represents a service to deploy resources inside a Kubernetes environment.
 type KubernetesDeployer struct {
-	binaryPath string
+	command string
 }
 
 // NewKubernetesDeployer initializes a new KubernetesDeployer service.
 func NewKubernetesDeployer(binaryPath string) *KubernetesDeployer {
+	command := path.Join(binaryPath, "kubectl")
+	if runtime.GOOS == "windows" {
+		command = path.Join(binaryPath, "kubectl.exe")
+	}
+
 	return &KubernetesDeployer{
-		binaryPath: binaryPath,
+		command: command,
 	}
 }
 
 // Deploy will deploy a Kubernetes manifest inside a specific namespace
 // it will use kubectl to deploy the manifest.
-func (deployer *KubernetesDeployer) Deploy(token, data string, namespace string) ([]byte, error) {
-	command := path.Join(deployer.binaryPath, "kubectl")
-	if runtime.GOOS == "windows" {
-		command = path.Join(deployer.binaryPath, "kubectl.exe")
+// kubectl uses in-cluster config.
+func (deployer *KubernetesDeployer) Deploy(name string, filePaths []string, prune bool) error {
+	if len(filePaths) == 0 {
```

Comment:
This function always runs kubectl with in-cluster config which has admin permission. It's a security hold.
As DeployRawConfig does,  it should run kubectl with the token from the request if there is any.

I'm not sure portainer allows non-admin users to deploy stacks or not. If the answer is not, the above comment is not necessary. But it's always good to use the token from request IMO.

Reply:
edge stacks are deployable by admin only. `Deploy` isn't called by an http request so there's no token.